Production_scripts/Social_Care/Care_Home.R

```diff
@@ -0,0 +1,417 @@
+      ch_discharge_date
+    )
+  )
+
+
+# Match with demographics data  ---------------------------------------
+
+# read in demographic data
+sc_demog <- readr::read_rds(get_sc_demog_lookup_path())
+
+# Get a list of confirmed valid Scottish postcodes from the SPD
+valid_spd_postcodes <- readr::read_rds(get_spd_path()) %>%
+  pull(pc7)
+
+matched_ch_data <- ch_clean %>%
+  left_join(sc_demog, by = c("sending_location", "social_care_id")) %>%
+  # Make the care home name more uniform
+  mutate(ch_name = clean_up_free_text(ch_name)) %>%
+  # correct postcode formatting
+  mutate(across(contains("postcode"), phsmethods::format_postcode),
+    # replace invalid postcode with NA
+    ch_postcode = na_if(ch_postcode, ch_postcode %in% valid_spd_postcodes)
+  )
+
+
+# Care Home Name lookup
+ch_name_lookup <-
+  # Care Home name lookup from the Care Inspectorate
+  # Previous contact Al Scrougal - Al.Scougal@careinspectorate.gov.scot
+  readxl::read_xlsx(get_slf_ch_name_lookup_path()) %>%
+  # Drop any Care Homes that were closed before 2017/18
+  filter(is.na(DateCanx) | DateCanx >= start_fy("1718")) %>%
+  rename(
+    ch_postcode = "AccomPostCodeNo",
+    ch_name_validated = "ServiceName"
+  ) %>%
+  select(
+    ch_postcode,
+    ch_name_validated,
+    DateReg,
+    DateCanx
+  ) %>%
+  # Standardise the postcode and CH name
+  mutate(
+    ch_postcode = phsmethods::format_postcode(ch_postcode),
+    ch_name_validated = clean_up_free_text(ch_name_validated)
+  ) %>%
+  # Merge any duplicates, and get the interval each CH name was active
+  group_by(ch_postcode, ch_name_validated) %>%
+  summarise(open_interval = interval(min(DateReg), replace_na(max(DateCanx), Sys.Date()))) %>%
+  # Find the latest date for each CH name
+  mutate(latest_close_date = max(int_end(open_interval))) %>%
+  ungroup() %>%
+  arrange(open_interval)
+
+# Generate some metrics for how the submitted names connect to the valid names
+ch_name_match_metrics <- matched_ch_data %>%
+  distinct(ch_postcode, ch_name) %>%
+  left_join(ch_name_lookup, by = c("ch_postcode")) %>%
+  drop_na() %>%
+  # Work out string distances between names for each postcode
+  mutate(
+    match_distance_jaccard = stringdist::stringdist(ch_name, ch_name_validated, method = "jaccard"),
+    match_distance_cosine = stringdist::stringdist(ch_name, ch_name_validated, method = "cosine"),
+    match_mean = (match_distance_jaccard + match_distance_cosine) / 2
+  ) %>%
+  # Drop any name matches which aren't very close
+  filter(match_distance_jaccard <= 0.25 | match_distance_cosine <= 0.3) %>%
+  group_by(ch_postcode, ch_name) %>%
+  # Identify the closest match in case there are multiple close matches
+  mutate(
+    min_jaccard = min(match_distance_jaccard, na.rm = TRUE),
+    min_cosine = min(match_distance_cosine, na.rm = TRUE),
+    min_match_mean = min(match_mean, na.rm = TRUE)
+  ) %>%
+  ungroup()
+
+no_postcode_match <- matched_ch_data %>%
+  anti_join(ch_name_lookup, by = "ch_postcode")
```

Comment:
I don't currently do anything with these, but probably could e.g. look for name matches (or close fuzzy matches) then 'correct' the postcode if there's a good match.